src/pcre2_compile.c

```diff
@@ -11096,7 +11142,7 @@ if ((re->overall_options & PCRE2_NO_START_OPTIMIZE) == 0)
   that disables this case.) */
 
   else if ((re->overall_options & PCRE2_ANCHORED) == 0 &&
-           is_startline(codestart, 0, &cb, 0, FALSE))
+           is_startline(codestart, 0, &cb, 0, FALSE, (optim_flags & PCRE2_OPTIM_DOTSTAR_ANCHOR) != 0))
```

Comment:
I think it looks better if the whole argument goes to the next line instead, should align with "codestart"

Reply:
Adjusted.